numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1704,15 +1996,24 @@ unpack_bits(PyObject *input, int axis)
         unsigned const char *inptr = PyArray_ITER_DATA(it);
         char *outptr = PyArray_ITER_DATA(ot);
 
-        for (index = 0; index < n_in; index++) {
-            unsigned char mask = 128;
+        if (out_stride == 1) {
+            for (index = 0; index < n_in; index++) {
+                memcpy(outptr, unpack_lookup[*inptr], 8);
+                outptr += 8;
+                inptr += in_stride;
+            }
+        }
+        else {
+            for (index = 0; index < n_in; index++) {
+                unsigned char mask = 128;
 
-            for (i = 0; i < 8; i++) {
-                *outptr = ((mask & (*inptr)) != 0);
-                outptr += out_stride;
-                mask >>= 1;
+                for (i = 0; i < 8; i++) {
```

Comment:
I wonder if it might be faster to use a local buffer and the lookup table, then copy the result to the destination?